### PR TITLE
Make things private that should be so

### PR DESCRIPTION
**Pull request by vagababov**
Make some constants and enums private.
The ones I found to be used elsewhere, I kept public.

/assign @markusthoemmes @yanweiguo

### CODE REVIEW

File: metrics/config.go [hunk 1]
```diff
@@ -40,32 +40,32 @@ const (
 	// The following keys are used to configure metrics reporting.
 	// See https://github.com/knative/serving/blob/master/config/config-observability.yaml
 	// for details.
-	AllowStackdriverCustomMetricsKey = "metrics.allow-stackdriver-custom-metrics"
+	allowStackdriverCustomMetricsKey = "metrics.allow-stackdriver-custom-metrics"
 	BackendDestinationKey            = "metrics.backend-destination"
-	ReportingPeriodKey               = "metrics.reporting-period-seconds"
+	reportingPeriodKey               = "metrics.reporting-period-seconds"
 	// Stackdriver client configuration keys
```
Comment (matzew) [inline metrics/config.go:46]: Are we fine w/ mixing private/public in one "block" ?

or should the be "ordered" (private/public and alphabetically?)

Comment (vagababov) [reply metrics/config.go:46]: :shrug: I can do that.

Comment (vagababov) [reply metrics/config.go:53]: ok sorted and reordered and split enum alone.

**Pull request merged**